File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public static char isAlphaNumeric(char ch) {
        if(ch >= 'a' && ch <= 'z') return ch;
        if(ch >= 'A' && ch <= 'Z') return (char)(ch+32);
        if(ch >= '0' && ch <= '9') return ch;
        return '@';
    }
    public boolean isPalindrome(String s) {
        int n = s.length();
        StringBuilder ans = new StringBuilder("");

        for(int i=0; i<n; i++) {
            char ch = s.charAt(i);
            char ch2 = isAlphaNumeric(ch);
            if(ch2 != '@') ans.append(ch2);
        }

        String str = ans.toString();

        int si = 0;
        int ei = str.length()-1;       

        while(si <= ei) {
            if(str.charAt(si) != str.charAt(ei)) return false;
            si++;
            ei--;
        }

        return true;
    }
}

File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isAlphaNumeric(char ch) {
        if(ch >= 'a' && ch <= 'z') return true;
        if(ch >= 'A' && ch <= 'Z') return true;
        if(ch >= '0' && ch <= '9') return true;

        return false;
    }

    bool isCapital(char ch) {
        if(ch >= 'A' && ch <= 'Z') return true;
        return false;
    }
    bool isPalindrome(string s) {
        string str = "";

        for(int i=0; i<s.length(); i++) {
            char ch = s[i];
            if(isAlphaNumeric(ch)) {
                if(isCapital(ch)) ch = tolower(ch);
                str += ch;
            }
        }

        int si = 0;
        int ei = str.length()-1;

        while(si <= ei) {
            if(str[si] != str[ei]) return false;
            si++;
            ei--;
        }
        return true;
    }
};